feeds/scheduler/scheduler.go

```diff
@@ -51,6 +63,29 @@ func (s *Scheduler) Poll(cutoff time.Time) ([]*feeds.Package, []error) {
 		}
 		packages = append(packages, result.packages...)
 		logger.WithField("num_processed", len(result.packages)).Print("processed packages")
+
+		previousResult, ok := s.previousPollResults[result.name]
+		nonZeroResults := len(previousResult.packages) > 0 && len(result.packages) > 0
+		if ok && nonZeroResults {
+			if !result.isOverlappingWith(s.previousPollResults[result.name]) {
+				s.eventHandler.DispatchEvent(events.LossyFeedEvent{
+					Feed: result.name,
+				})
+			}
+		}
+		s.previousPollResults[result.name] = result
 	}
 	return packages, errs
 }
+
+// Checks whether there is an overlap in package creation date between a result
+// and a previous result. This assumes that pollResult.packages is sorted by
+// CreatedDate in order of most recent first.
+func (r pollResult) isOverlappingWith(previousResult pollResult) bool {
```

Comment:
With the current static polling rate and how it is then used to create a cutoff point, isn't the current expectation here that there will be no overlaps (as they've already been filtered by the cutoff when the Package structs are initialised?) between each scheduled poll, assuming singular instance?

Reply:
Indeed, I think we will need to move the cutoff calculation outside of the feeds to allow for these feeds to be stored and compared.

The filter of packages with cutoff can be done prior to the return of `Poll()`

Reply:
This comment is null now, as it's applied pre-cutoff filtering